dawn/src/dawn/Optimizer/PassValidation.cpp

```diff
@@ -0,0 +1,67 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
```

Comment:
validation now has a bunch of ways to do things:
-Throwing from `IntegrityChecker`
- `DAWN_ASSERT_MESSAGE`s from the grid-checkers
- returning `nullptrs` in validation

I think we should make an effort do unify things.
The idea of the `DAWN_ASSERT`s was that they could be disabled. What is the advantage of throwing?

-

Reply:
Standardized PassValidation to throw exceptions. Benefits are that we can throw/catch errors and aggregate them in the diagnostics, without halting execution and gather as many errors as possible.

Reply:
Reverted grid/dimension type checks to asserts per PR #583